packages/typescript/ai/tests/debug-logging-chat.test.ts

```
@@ -0,0 +1,280 @@
+/**
+ * Create a mock adapter whose chatStream throws once iteration begins.
+ * Exercises the error path through the chat pipeline.
+ */
+function createFailingMockAdapter(
+  message = 'mock adapter failure',
+): ReturnType<typeof createMockAdapter> {
+  return createMockAdapter({
+    chatStreamFn: () =>
+      (async function* (): AsyncIterable<StreamChunk> {
+        throw new Error(message)
+      })(),
```

ESLint

[error] This generator function does not have 'yield'.

(require-yield)